File: lighthouse/src/main/java/com/automation/center/lighthouse/model/Metric.java
package com.automation.center.lighthouse.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Entity(name = "metric")
@Getter
@Setter
@ToString
public class Metric {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name")
    private String name;
    @Column(name = "code")
    private String code;

    @ManyToMany(mappedBy = "metrics")
    @ToString.Exclude
    private List<SuiteItem> suiteItems = new ArrayList<>();
}

File: lighthouse/src/main/java/com/automation/center/lighthouse/model/Metric.kt
package com.automation.center.lighthouse.model

import jakarta.persistence.*

@Entity
@Table(name = "metric")
class Metric() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var name: String? = null
    var code: String? = null
    var jsonPath: String? = null


    @OneToMany(mappedBy = "metric", cascade = [CascadeType.ALL], orphanRemoval = true)
    var results: List<Result> = ArrayList()


    @ManyToMany
    @JoinTable(
        name = "suite_metric",
        joinColumns = [JoinColumn(name = "metric_id")],
        inverseJoinColumns = [JoinColumn(name = "suite_id")]
    )
    var suites: List<Suite> = ArrayList()

    constructor(id: Long) : this() {
        this.id = id
    }

    constructor(name: String, code: String) : this() {
        this.name = name
        this.code = code
    }
}
